src/Adapter/Product/Combination/Update/CombinationStockUpdater.php

```diff
@@ -155,6 +166,13 @@ private function updateStockAvailable(Combination $combination, CombinationStock
         // save movement only after stockAvailable has been updated
         if ($stockModification) {
             $this->saveMovement($stockAvailable, $stockModification);
+
+            // Update reserved and physical quantity for this stock
+            $this->stockAvailableRepository->updatePhysicalProductQuantity(
+                new StockId((int) $stockAvailable->id),
+                new OrderStateId((int) $this->configuration->get('PS_OS_ERROR', null, ShopConstraint::shop((int) $stockAvailable->id_shop))),
+                new OrderStateId((int) $this->configuration->get('PS_OS_CANCELED', null, ShopConstraint::shop((int) $stockAvailable->id_shop)))
+            );
```

Comment:
```suggestion

            // Update reserved and physical quantity for this stock
            $this->stockAvailableRepository->updatePhysicalProductQuantity(
                new StockId((int) $stockAvailable->id),
                new OrderStateId($this->configuration->getInt('PS_OS_ERROR', null, ShopConstraint::shop((int) $stockAvailable->id_shop))),
                new OrderStateId($this->configuration->getInt('PS_OS_CANCELED', null, ShopConstraint::shop((int) $stockAvailable->id_shop)))
            );
```

Reply:
I can't do this all the getters from the ParameterBag don't implement the third ShopConstraint parameter As long as we don't update the interface with this parameter we are kind of stuck and need to use the Adapter service

Besides `getInt` `getBool` and so on are not part of the `ConfigurationInterface` either so your two comments here are a bit contradictory We will have to one day update this interface and ideally force the shop constraint parameter which would remove any ambiguity related to the multishop feature but it's a big breaking change which we can't afford yet